Added the Set GP Layer Attributes and Set GP Stroke Attributes nodes.

3DSinghVFX opened the pull request:
I have added the _Set GP Layer Attributes_ and _Set GP Stroke Attributes_ nodes which allow to set all the attributes together of a Layer and Stroke,
![Screenshot from 2020-02-11 01-21-42](https://user-images.githubusercontent.com/30294746/74184511-ed526380-4c6c-11ea-8463-a041ac167859.png)

_More discussion for these nodes is here: https://github.com/JacquesLucke/animation_nodes/pull/1307#issuecomment-584209257_

OmarEmaraDev commented:
The input attributes should be independent of the `useLayerList`. In other words, we should allow a mix between single and list attributes.

You may also use a virtual list for layers and strokes. And you shouldn't create virtual lists if they wre not going to be used.

Also, remove the set nodes we added in #1307.

3DSinghVFX commented:
> The input attributes should be independent of the `useLayerList`. In other words, we should allow a mix between single and list attributes.

It is added.

> You may also use a virtual list for layers and strokes. And you shouldn't create virtual lists if they wre not going to be used.

It is added.
 
> Also, remove the set nodes we added in #1307.

Removed.

OmarEmaraDev commented on animation_nodes/nodes/gpencil/set_gp_layer_attributes.py:
```diff
@@ -0,0 +1,78 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... data_structures import GPLayer, VirtualPyList, VirtualDoubleList, VirtualLongList
```
This import is not needed.

3DSinghVFX replied:
Done.

OmarEmaraDev commented on animation_nodes/nodes/gpencil/set_gp_layer_attributes.py:
```diff
@@ -0,0 +1,78 @@
+        isBlendMode = s[2].isUsed
+        isOpacity = s[3].isUsed
+        isPassIndex = s[4].isUsed
+        if self.useLayerList or self.useNameList or self.useBlendModeList or self.useOpacityList or self.usePassIndexList:
```
Use the built-in `any()` function.

3DSinghVFX replied:
Done.

OmarEmaraDev commented on animation_nodes/nodes/gpencil/set_gp_layer_attributes.py:
```diff
@@ -0,0 +1,78 @@
+            ("Opacity", "opacities"), ("Opacities", "opacities")), value = 1, hide = True)
+        self.newInput(VectorizedSocket("Integer", "usePassIndexList",
+            ("Pass Index", "passIndices"), ("Pass Indices", "passIndices")), value = 0, hide = True)
+        self.newOutput(VectorizedSocket("GPLayer", ["useLayerList", "useNameList", "useBlendModeList",
```
Put a new line before `self.newOutput`. Try to organize this a bit more, maybe:
```python
self.newOutput(VectorizedSocket("GPLayer",
    ["useLayerList", "useNameList", "useBlendModeList", "useOpacityList", "usePassIndexList"],
    ("Layer", "outLayer"), ("Layers", "outLayers")))
```

3DSinghVFX replied:
Done.

3DSinghVFX commented:
Is it possible to set that in the _newInput_?

OmarEmaraDev commented:
Yes. You can pass `minValue` if I remember correctly.

3DSinghVFX commented:
Okay.

OmarEmaraDev commented:
Opacity needs a maximum of 1 as well.

3DSinghVFX commented:
It is done.

3DSinghVFX commented:
Should I update the CHANGELOG.md?

OmarEmaraDev commented:
Yes. Please do that before submitting the pull request in the future.

3DSinghVFX commented:
> Yes. Please do that before submitting the pull request in the future.

Okay.
